tests/resources/test_record_file_permissions.py

```diff
@@ -72,6 +72,99 @@ def record_w_restricted_file(client, headers, running_app, minimal_record, users
     return recid
 
 
+def _add_to_community(record, community, service, db):
+    """Add a record to a community."""
+    record.parent.communities.add(community._record, default=False)
+    record.parent.commit()
+    record.commit()
+    db.session.commit()
+    service.indexer.index(record, arguments={"refresh": True})
+    return record
+
+
+def test_community_access_restricted_files(
+    test_user,
+    client,
+    uploader,
+    record_community,
+    full_record,
+    community_owner,
+    headers,
+    community,
+    minimal_record,
+    db,
+):
+    """Test the access to restricted files of a record for record owner, community owner and a simple user ."""
+    service = current_rdm_records_service
+    record_owner = uploader.login(client)
+
+    # publish a record with restricted files
+    minimal_record["access"]["files"] = "restricted"
+    recid = create_record_w_file(record_owner, minimal_record, headers)
+
+    req_item = service.read(uploader.identity, recid)
+    _add_to_community(req_item._record, community, service, db)
```

Comment:
why not using `record = record_community.create_record()`, which will already create a record in a community, without duplicating the code above `_add_to_community`?

Reply:
Because it creates and publishes a new record without files inside(I'm adding files in `create_record_w_file`)